storage/src/snapshot.rs

```diff
@@ -215,8 +219,9 @@ impl StateRead for Snapshot {
                         let k = std::str::from_utf8(key_preimage.as_ref())
                             .expect("saved jmt keys are utf-8 strings")
                             .to_string();
+
                         let v = self2
-                            .get_jmt(k.as_bytes().into())?
+                            .get_jmt(KeyHash::with::<Sha256>(k.clone()))?
```

Comment:
wait, why doesn't `.into()` work to construct key hashes any more?

Reply:
we removed it with `0.4.0`, I think to make space for the generic hasher, see:

https://github.com/penumbra-zone/jmt/commit/2475dfb7b73bf16de518c2de34f10e00cd4fa909#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759L160

Reply:
Ah, hmm, that seems unfortunate. I think it'd be better if there was a `From` impl, but not really relevant for this PR.

Reply:
perhaps something like:

```rust
pub struct KeyHash<H: SimpleHasher>(pub [u8; 32], PhantomData<H>);
```

?